web/packages/teleport/src/Integrations/IntegrationList.tsx

```diff
@@ -0,0 +1,180 @@
+/**
+ * Copyright 2023 Gravitational, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import React from 'react';
+import styled from 'styled-components';
+
+import { Box, Flex, Image } from 'design';
+import awsIcon from 'design/assets/images/icons/aws.svg';
+import slackIcon from 'design/assets/images/icons/slack.svg';
+import Table, { Cell } from 'design/DataTable';
+import { MenuButton, MenuItem } from 'shared/components/MenuAction';
+
+import { IntegrationCode } from 'teleport/services/integrations';
+
+import type { Integration, Plugin } from 'teleport/services/integrations';
+
+type Props<Integrations> = {
+  list: Integrations[];
+  onDelete(i: Integrations): void;
+};
+
+type Integrations = Integration | Plugin;
+
+export function IntegrationList(props: Props<Integrations>) {
+  return (
+    <Table
+      pagination={{ pageSize: 20 }}
+      isSearchable
+      data={props.list}
+      columns={[
+        {
+          key: 'kind',
+          headerText: 'Integration',
+          isSortable: true,
+          render: item => <IconCell item={item} />,
+        },
+        {
+          key: 'details',
+          headerText: 'Details',
+        },
+        {
+          key: 'statusCodeText',
+          headerText: 'Status',
+          isSortable: true,
+          render: item => <StatusCell item={item} />,
+        },
+        {
+          altKey: 'options-btn',
+          render: item => (
+            <ActionCell
+              onDelete={props.onDelete ? () => props.onDelete(item) : null}
+            />
+          ),
+        },
+      ]}
+      emptyText="No Results Found"
+    />
+  );
+}
+
+const StatusCell = ({ item }: { item: Integrations }) => {
+  let success;
+  let warning;
+  let error;
+
+  if (item.resourceType === 'plugin') {
+    if (item.statusCode === 'Running') {
+      success = true;
+    } else if (
+      item.statusCode === 'Unauthorized' ||
+      item.statusCode === 'Unknown error'
+    ) {
+      error = true;
+    } else if (item.statusCode === 'Bot not invited to channel') {
+      warning = true;
+    }
+  } else {
+    // default to integration resource type.
+    if (item.statusCode === IntegrationCode.Running) {
+      success = true;
+    } else if (item.statusCode === IntegrationCode.Paused) {
+      warning = true;
+    } else if (item.statusCode === IntegrationCode.Error) {
+      error = true;
+    }
+  }
```

Comment:
What do you think about abstracting this into a function that can return an enum for the status?

Something like

```typescript
enum Status {
  Success,
  Warning,
  Error,
}

function getStatus(item: Integrations) {
  if (item.resourceType === 'plugin') {
    switch (item.statusCode) {
      case 'Running':
        return Status.Success;

      case 'Unauthorized':
      case 'Unknown error':
        return Status.Error;

      case 'Bot not invited to channel':
        return Status.Warning;
    }

    return;
  }

  switch (item.statusCode) {
    case IntegrationCode.Running:
      return Status.Success;

    case IntegrationCode.Paused:
      return Status.Warning;

    case IntegrationCode.Error:
      return Status.Error;
  }
}
```

Which would then simplify `StatusCell` to 

```typescript jsx
const StatusCell = ({ item }: { item: Integrations }) => {
  const status = getStatus(item);

  return (
    <Cell>
      <Flex alignItems="center">
        <StatusLight status={status} />
        {item.statusCodeText}
      </Flex>
    </Cell>
  );
};
```

and `StatusLight` receives 1 prop instead of n props for each state

```typescript
const StatusLight = styled(Box)`
  border-radius: 50%;
  margin-right: 4px;
  width: 8px;
  height: 8px;
  background-color: ${({ status, theme }) => {
    if (status === Status.Success) {
      return theme.colors.success;
    }
    if (status === Status.Error) {
      return theme.colors.error.light;
    }
    if (status === Status.Warning) {
      return theme.colors.warning;
    }
    return theme.colors.grey[300]; // Unknown
  }};
`;
```